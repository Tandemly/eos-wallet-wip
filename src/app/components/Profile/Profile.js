import * as React from "react";
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import cx from "classnames";
import css from "./styles.module.scss";

// TODO styles
const Profile = ({
  image: {
    url: src,
  },
  currentLocation,
  name,
  status,
}) => (
  <div className={cx("box", css.profile)}>
    <div className="level">
      <div className="level-left">
        <div className="level-item">
          <img
            src={src}
            className={css.thumbnail}
            alt={name}
          />
        </div>
        <div className="level-item">
          <div>
            <p className={cx("heading is-6", css.heading)}>{name}</p>
            <p className={cx("title", css.title)}>{currentLocation}</p>
          </div>
        </div>
      </div>
    </div>
    <div className="u-mt2">
      <p>{status}</p>
    </div>
  </div>
);

export default Profile;
