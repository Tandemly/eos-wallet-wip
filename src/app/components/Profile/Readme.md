Profile example:

```js
<Background className="prefrences-form-example u-p6">
  <Profile
    name="Beth"
    currentLocation="Creve Coeur, MO"
    image={{ url: "https://avatarfiles.alphacoders.com/696/69632.jpg" }}
    status="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Duis a elit pulvinar, varius mauris non, condimentum tellus. Maecenas vitae ultrices diam. Lorem ipsum dolor sit amet, consectetur elit."
  />
</Background>
```
