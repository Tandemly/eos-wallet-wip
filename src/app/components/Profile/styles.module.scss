@import "../../styles/custom_variables.scss";

.profile {
  .thumbnail {
    border: solid 8px $orange;
    width: 150px;
    height: 150px;
    border-radius: 75px;
  }
  .heading {
    font-size: 22px;
    font-weight: bold;
    text-transform: capitalize;
  }
  .title {
    font-size: 18px;
    color: $white;
  }
}