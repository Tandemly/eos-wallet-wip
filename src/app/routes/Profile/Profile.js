import * as React from "react";
import { Helmet } from 'react-helmet';
import Balance from "containers/Balance";
import Transactions from "containers/Transactions";

const Profile = ({ account_name }) => (
  <div>
    <Helmet>
      <title>Profile</title> {/* TODO: Show name of user here instead */}
    </Helmet>

    <div className="container-full">
      <div className="row">
        <div className="col-12">
          <h2>Username</h2>
        </div>
      </div>
      <div className="section">
        <div className="row">
          <div className="col-12 col-lg-8 d-sm-inline-flex flex-row about">
            <div className="profile-pic" />
            <div className="profile-meta mt-2 ml-0 ml-sm-4">
              <div className="username">Display Name</div>
              <div className="location">Location</div>
              <div className="website">website</div>
              <div className="bio mt-1">Wild-man, Gardener, Encourager, Homesteader, Lyricist, Deep Thinker, Papa, Humorist, Husband
              </div>
            </div>
          </div>
          <div className="col-12 col-lg-4 mt-lg-0 mt-4">
            <div className="module snapshot text-lg-right text-left">
              <Balance
                account_name={account_name}
              />
            </div>
          </div>
        </div>
      </div>

      <div className="d-md-flex justify-content-between items-center">
        <div>
          <h3>Transaction History</h3>
        </div>
        <div className="items-center mb-2 mb-md-0">
          <div className="module-search">
            <input
              className="form-control"
              type="text"
            />
          </div>
        </div>
      </div>
      <div className="row">
        <div className="col-12">
          <Transactions
            account_name={account_name}
            count={5}
          />
        </div>
      </div>
    </div>
  </div>
);

export default Profile;
