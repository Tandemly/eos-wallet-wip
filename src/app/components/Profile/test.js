import * as React from "react";
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router';
import Profile from './';

describe('<Profile />', () => {
  it('renders without crashing', () => {
    const staticContext = {};
    const profile = {
      image: {
        url: 'src',
      },
      currentLocation: 'Friday Harbor, WA',
      name: 'Rick Sanchez',
      status: 'active',
    };

    const tree = renderer.create(
      <StaticRouter location="/" context={staticContext}> 
        <Profile {...profile} />
      </StaticRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
})

