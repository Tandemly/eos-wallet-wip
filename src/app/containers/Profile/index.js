import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import Profile from "components/Profile";

// TODO genericize this to network call to specific user
const mapStateToProps = ({ login: { user } }) => ({
  name: user.display_name || user.account_name,
  image: { url: user.image_url }, 
});

const Container = connect(
  mapStateToProps,
)(Profile);

export default withRouter(Container);
