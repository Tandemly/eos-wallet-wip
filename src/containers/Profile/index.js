import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import Profile from "components/Profile";

// TODO genericize this to network call to specific user
const mapStateToProps = ({ login: { user: { name, picture }}}) => ({
  name,
  image: { url: picture },
});

const Container = connect(
  mapStateToProps,
)(Profile);

export default withRouter(Container);
