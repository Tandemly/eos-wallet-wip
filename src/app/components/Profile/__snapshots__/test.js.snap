// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> renders without crashing 1`] = `
<div
  className="box"
>
  <div
    className="level"
  >
    <div
      className="level-left"
    >
      <div
        className="level-item"
      >
        <img
          alt="Rick Sanchez"
          className={undefined}
          src="src"
        />
      </div>
      <div
        className="level-item"
      >
        <div>
          <p
            className="heading is-6"
          >
            Rick Sanchez
          </p>
          <p
            className="title"
          >
            Friday Harbor, WA
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="u-mt2"
  >
    <p>
      active
    </p>
  </div>
</div>
`;
